autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -629,7 +634,14 @@ def predict_model(self, X, model, model_pred_proba_dict=None):
         if isinstance(model, str):
             model = self.load_model(model)
         X = self.get_inputs_to_model(model=model, X=X, model_pred_proba_dict=model_pred_proba_dict, fit=False)
-        return model.predict(X=X, preprocess=False)
+        y_pred = model.predict(X=X, preprocess=False)
+        if self.regress_preds_asprobas and model.problem_type == REGRESSION:  # Convert regression preds to classes (during distillation)
```

Comment:
No because non-distilled models should never have `model.problem_type == REGRESSION` if the original problem type was classification (which is the only case where `self.regress_preds_asprobas` would've been set = True).